core/esmf-aspect-model-aas-generator/src/main/java/org/eclipse/esmf/aspectmodel/aas/AspectModelAasVisitor.java

```diff
@@ -734,6 +745,32 @@ public Environment visitEnumeration( final Enumeration enumeration, final Contex
       return context.environment;
    }
 
+   /**
+    * Transforms a given enumValue string to a specific format.
+    * The transformation is done by removing spaces and special characters
+    * from the input string, then appending the first 8 characters of the
+    * sha256 hash of the cleaned string to the provided prefix.
+    *
+    * @param enumValue the input string to be transformed
+    * @return the transformed string in the format "_role[8_characters_of_hash]"
+    * @throws NoSuchAlgorithmException if the SHA-256 algorithm is not available
+    */
+   private String transformEnumerationValue( final String enumValue ) throws NoSuchAlgorithmException {
```

Comment:
How would the mapping work in the opposite direction with this approach?

Reply:
I think, that I have to add separate method in the opposite direction.

Thanks!